vision/segmentation/er_ccl_clustering.go

```diff
@@ -0,0 +1,274 @@
+package segmentation
+
+import (
+	"context"
+	"math"
+
+	"github.com/golang/geo/r3"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/components/camera"
+	pc "go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/utils"
+	"go.viam.com/rdk/vision"
+)
+
+// ErCCLConfig specifies the necessary parameters to apply the
+// connected components based clustering algo.
+type ErCCLConfig struct {
+	resource.TriviallyValidateConfig
+	MinPtsInPlane    int       `json:"min_points_in_plane"`
+	MinPtsInSegment  int       `json:"min_points_in_segment"`
+	MaxDistFromPlane float64   `json:"max_dist_from_plane_mm"`
+	NormalVec        r3.Vector `json:"ground_plane_normal_vec"`
+	AngleTolerance   float64   `json:"ground_angle_tolerance_degs"`
+	ClusteringRadius int       `json:"clustering_radius"`
+	Beta             float64   `json:"beta"`
```

Comment:
I think we need to come up with a better name for beta. something like "clustering_granularity" or "clustering_strictness"? Beta sets how fast the threshold decays -- higher beta make it harder to pass the threshold and be assigned to the same cluster, so you end up creating more clusters if beta is very high.